pkg/apis/crd/v1alpha1/types.go

```diff
@@ -709,3 +709,136 @@ type ExternalNodeList struct {
+}
+
+// BundlePhase defines the phase in which a SupportBundle is.
+type BundlePhase string
+
+const (
+	// BundlePending means the SupportBundle has been accepted by the system, but it has not been processed by Antrea.
+	BundlePending BundlePhase = "Pending"
+	// BundleProcessing means the SupportBundle has been observed by Antrea and is being processed.
+	BundleProcessing BundlePhase = "Processing"
+	// BundleProcessed means the SupportBundle has been processed by all Nodes or ExternalNodes it requires.
+	BundleProcessed BundlePhase = "Processed"
+	// BundleFailed means the SupportBundle has been marked by Antrea when time is up. This is used when any of the
+	// required Node or ExternalNode fails to process the request.
+	BundleFailed BundlePhase = "Failed"
+)
+
+// BundleFailedReason defines the reason why a SupportBundle is marked as Failed.
+type BundleFailedReason string
+
+const (
+	// Timeout means any of the required Nodes or ExternalNodes fails to process the SupportBundle request
+	// in the given timeout.
+	Timeout BundleFailedReason = "Timeout"
+	// ControllerError means an error occurs on Antrea Controller which breaks processing the SupportBundle.
+	ControllerError BundleFailedReason = "Controller error"
+	// UnknownError means an unknown error occurs in Antrea when processing the SupportBundle
+	UnknownError BundleFailedReason = "Unknown"
+)
+
+type BundleStatus struct {
+	Phase BundlePhase `json:"phase"`
+	// The generation observed by Antrea.
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// The number of Nodes and ExternalNodes that have processed the SupportBundle.
+	CurrentNodes int32 `json:"currentNodes"`
+	// The total number of Nodes and ExternalNodes that should process the SupportBundle.
+	DesiredNodes int32 `json:"desiredNodes"`
+	// The reason why the SupportBundle phase is marked as Failed.
+	FailedReason BundleFailedReason `json:"failedReason,omitempty"`
+}
+
+type BundleNodes struct {
+	// List the names of certain Nodes which are expected to collect and upload
+	// bundle files.
+	// +optional
+	NodeNames []string `json:"nodeNames,omitempty"`
+	// Select certain Nodes which match the label selector.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+}
+
+type BundleExternalNode struct {
+	Namespace string
+	Nodes     BundleNodes
+}
+
+// BundleUploadScheme defines the scheme that the BundleFileServer uses in the url to upload bundle files.
+type BundleUploadScheme string
+
+const (
+	HTTPScheme  BundleUploadScheme = "http"
+	HTTPSScheme BundleUploadScheme = "https"
+)
+
+// BundleUploadVerb defines the verb to upload bundle files to the URL provided by the BundleFileServer.
+type BundleUploadVerb string
+
+const (
+	PostBundle BundleUploadVerb = "post"
+	PutBundle  BundleUploadVerb = "put"
+)
+
+// BundleFileServer defines how to upload bundle files.
+type BundleFileServer struct {
+	Host   string             `json:"host"`
```

Comment:
Question - why we do not use a single URL string, but have both Host and Path?

Reply:
Do you think we should have separate fields for different protocols like:
```
type BundleFileServer struct {
         HttpServer string
         FtpServer string
...
}
```

In your current approach, how we define port number?

And in general, I would not introduce complexity for potential features. If we do not have a plan to support other protocols, let us just define API for HTTPS.